app/models/submission.rb

```diff
@@ -45,28 +45,110 @@ def archived
     self.archived?
   end
 
+  # Validate each submitted field against its question type
   def validate_custom_form
-    @valid_form_condition = false
-
-    questions = form.questions
-
     # Isolate questions that were answered
     answered_questions = attributes.select { |_key, value| value.present? }
+
     # Filter out all non-question attributes
+    answered_questions.delete('id')
+    answered_questions.delete('uuid')
     answered_questions.delete('touchpoint_id')
     answered_questions.delete('form_id')
     answered_questions.delete('user_agent')
+    answered_questions.delete('hostname')
     answered_questions.delete('page')
     answered_questions.delete('ip_address')
     answered_questions.delete('language')
     answered_questions.delete('referer')
+    answered_questions.delete('aasm_state')
+    answered_questions.delete('spam_score')
+    answered_questions.delete('created_at')
+    answered_questions.delete('updated_at')
+
+    # Ensure only requested fields are submitted
+    expected_submission_fields = form.questions.collect(&:answer_field) + ["location_code"]
+    actual_submission_fields = answered_questions.keys
+    unexpected_fields = actual_submission_fields - expected_submission_fields
+    errors.add(:base, :invalid, message: "received invalid submission field(s): #{unexpected_fields.to_sentence}") if unexpected_fields.present?
+
+    # Assess each provided answer field against question-specific validations
+    form.questions.each do |question|
+      provided_answer = answered_questions[question.answer_field]
+
+      if provided_answer.nil? && question.is_required
+        errors.add(question.answer_field.to_sym, :blank, message: 'is required')
+      end
+
+      unless form.enforce_new_submission_validations || ENV.fetch("ENFORCE_NEW_SUBMISSION_VALIDATIONS", false)
```

Comment:
I'm probably missing some subtleties here but what would be wrong with validating new submissions to an existing form?

Reply:
using this as a feature flag (temporarily).
some data will needs to be migrated eg: "yes" and "no" to 1 / 0.
the feature flag is intended to give a little flexibility here